### DISCUSSION TITLE
Gtk development resources

### DISCUSSION THREAD

**Discussion by gloopsies** [flxzt/rnote#108] [Development]
I wanted to start developing with gtk, but I could never find good resources/examples/anything that can help me learn anything about gtk and how to use it. Do you have any recommended resources to learn from?

**Comment by flxzt** [comment]
A couple of resources helped me a lot:  
- The [rust gtk book](https://gtk-rs.org/gtk4-rs/git/book/)
- The general API documentations: [gtk-rs bindings](https://gtk-rs.org/gtk4-rs/stable/latest/docs/gtk4/) and the [C API](https://docs.gtk.org/) which is useful when you want to look up properties, the class hierarchy, ... Its sometimes nicer than the binding documentation
- The [Gnome developer docs](https://developer.gnome.org/documentation/index.html) which explain concepts like actions very well I think. There is also the Gnome [design guidelines](https://developer.gnome.org/hig/), but its not necessary at first I think, especially for an already existing application.
- there is the [rust gnome matrix channel](https://matrix.to/#/#rust:gnome.org) which is very helpful if you ask well informed questions. There is also a [discourse](https://discourse.gnome.org/tag/rust), although I haven't used it personally.
